core/src/main/clojure/xtdb/expression/list.clj

```diff
@@ -0,0 +1,26 @@
+(ns xtdb.expression.list
```

Comment:
let's have some tests for this - it's testable in its own right and there's a number of cases to consider:

- happy case
- expr that returns nil
- expr that doesn't return a list
- expr that _might_ return a list, but can potentially return other things as well